src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -81,12 +87,14 @@ public class StoreService : IStoreService
             {
                 _dicomInstanceEntries = instanceEntries;
                 _requiredStudyInstanceUid = requiredStudyInstanceUid;
+                IList<CustomTagEntry> customTagEntries = await _customTagStore.GetCustomTagsAsync(path: null, cancellationToken: cancellationToken);
```

Comment:
You should consider using IReadonlyCollection here. These lists need to be immutable.